include/stl2/detail/iterator/move_iterator.hpp

```diff
@@ -90,6 +91,33 @@ STL2_OPEN_NAMESPACE {
 				static_cast<void>(++current_)
 			)
 
+			// Experimental support for move_iterator post-increment
+			using __post_inc_t = decltype(current_++);
+			Readable{R}
+			struct __proxy {
+				R __tmp;
+				decltype(auto) operator*()
+				STL2_NOEXCEPT_RETURN(
+					__stl2::iter_move(__tmp)
+				)
+				friend decltype(auto) iter_move(const __proxy& that)
+				STL2_NOEXCEPT_RETURN(
+					__stl2::iter_move(that.__tmp)
+				)
+			};
+
+			STL2_CONSTEXPR_EXT cursor post_increment()
+			noexcept(noexcept(cursor{cursor{__stl2::exchange(current_, __stl2::next(current_))}}))
+			requires Same<__post_inc_t, I>() {
+				return cursor{__stl2::exchange(current_, __stl2::next(current_))};
```

Comment:
The critical feature here is that `next(foo)` stomps on `*foo` despite operating on a copy of `foo`, which is reasonable for input iterators since they are not required to be dereferenceable after any copy is incremented.

>what if we nuked that overload of `post_increment()` and only kept the one with the proxy that is implemented in terms of post-fix increment?

I could still implement the exact same iterator without using `basic_iterator`.